src/pk/ecc/ecc_ssh_ecdsa_encode_name.c

```diff
@@ -0,0 +1,57 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+#include "tomcrypt_private.h"
+
+/**
+   @file ecc_ssh_ecdsa_encode_name.c
+   Curve/OID to SSH+ECDSA name string mapping per RFC5656
+   Russ Williams
+*/
+
+int ecc_ssh_ecdsa_encode_name(char *buffer, unsigned long buflen, const ecc_key *key)
+{
+   int err;
+
+   /* Check for three named curves: nistp256, nistp384, nistp521 */
+   if (ecc_cmp_oid_str("1.2.840.10045.3.1.7", key) == CRYPT_OK) {
+      /* nistp256 - secp256r1 - OID 1.2.840.10045.3.1.7 */
+
+      if (buflen <= 19) { err = CRYPT_BUFFER_OVERFLOW; goto error; }
+      strcpy(buffer, "ecdsa-sha2-nistp256");
```

Comment:
clang-tidy complains here:
```
src/pk/ecc/ecc_ssh_ecdsa_encode_name.c:32:7: error: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119 [clang-analyzer-security.insecureAPI.strcpy,-warnings-as-errors]
      strcpy(buffer, "ecdsa-sha2-nistp384");
      ^
```
So perhaps use something else than `strcpy` e.g. `strncpy` / `XMEMCPY`

Reply:
Replaced with snprintf() and combined with buffer length checking, so no more magic numbers.